test-context/src/subset.rs

```diff
@@ -0,0 +1,90 @@
+use serde_json::Value;
+
+pub trait ContainsSubset {
+    // Returns true if the value is a subset of the receiver.
+    fn contains_subset(&self, value: Value) -> bool;
+    // Returns true if the value a deep subset of the receiver.
+    fn contains_deep_subset(&self, value: Value) -> bool;
+}
+
+impl ContainsSubset for Value {
+    fn contains_subset(&self, value: Value) -> bool {
+        match (self, &value) {
+            (Value::Object(src), Value::Object(subset)) => subset
+                .iter()
+                .all(|(k, v)| src.get(k).is_some_and(|x| x == v)),
+
+            (Value::Array(src), Value::Array(subset)) => {
+                subset.iter().all(|v| src.iter().any(|x| x == v))
+            }
+
+            _ => value == *self,
+        }
+    }
+
+    fn contains_deep_subset(&self, subset: Value) -> bool {
+        match (self, &subset) {
+            (Value::Object(src), Value::Object(tgt)) => tgt.iter().all(|(k, v)| {
+                src.get(k)
+                    .is_some_and(|x| x.contains_deep_subset(v.clone()))
+            }),
+
+            (Value::Array(src), Value::Array(subset)) => subset
+                .iter()
+                .all(|v| src.iter().any(|x| x.contains_deep_subset(v.clone()))),
+
+            _ => subset == *self,
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::subset::ContainsSubset;
+    use serde_json::json;
+
+    #[test]
+    fn test_is_subset() {
+        // actual can have additional fields
+        let actual = json!({
+            "relationship": "PackageOf",
+            "other": "test",
+        });
+        assert!(actual.contains_subset(json!({
+                "relationship": "PackageOf",
+        })));
+
+        // case where an expected field does not match
+        let actual = json!({
+            "relationship": "PackageOf",
+            "other": "test",
+        });
+        assert!(!actual.contains_subset(json!({
+            "relationship": "bad",
+        })));
+
+        // case where an expected field is missing
+        let actual = json!({
+            "relationship": "PackageOf",
+            "other": "test",
+        });
+        assert!(!actual.contains_subset(json!({
+                "name": "SATELLITE-6.15-RHEL-8",
+        })));
+    }
+
+    #[test]
+    fn test_array_subset() {
+        // actual can have additional fields
+        let actual = json!([1, 2, 3]);
+        assert!(actual.contains_subset(json!([2])));
+
+        // other values can be interleaved.
+        let actual = json!([1, 2, 3]);
+        assert!(actual.contains_subset(json!([1, 3])));
+
+        // case where a value is missing
+        let actual = json!([1, 2, 3]);
+        assert!(!actual.contains_subset(json!([0])));
```

Comment:
Maybe "strict" or "fuzzy" or "partial" are better descriptors than "deep". The crucial idea is that we need both a way to test for an explicit match and a way to ask "are all of these in the other one?"

I think this test is more accurate thusly:
```suggestion
        let actual = json!([1, 2, 3]);

        assert!(actual.contains_subset(json!([1, 2, 3])));
        assert!(!actual.contains_subset(json!([2])));

        assert!(actual.contains_deep_subset(json!([2])));
        assert!(actual.contains_deep_subset(json!([1, 3])));
        assert!(!actual.contains_deep_subset(json!([0])));
```

But again, I'm not sure "deep" is the right word. We should always recurse through a recursive structure.

Reply:
Yeah. I'd be happy with any of those options.